src/CSV.jl

```diff
@@ -704,34 +758,47 @@ end
         if user(T)
             if !options.strict
                 options.silencewarnings || warning(Int64, buf, pos, tlen, code, row, col)
-                @inbounds typecodes[col] = INT | MISSING
-                @inbounds tape[tapeidx] = MISSING_BIT
+                @inbounds typecodes[col] = T | MISSING
+                @inbounds tape[row] = uint64(INT_SENTINELS[intsentinel.x])
             else
                 stricterror(Int64, buf, pos, tlen, code, row, col)
             end
         else
             y, code, vpos, vlen, tlen = Parsers.xparse(Float64, buf, pos, len, options)
             if code > 0
-                @inbounds tape[tapeidx + 1] = uint64(y)
-                @inbounds typecodes[col] = FLOAT | (missingtype(T) ? MISSING : EMPTY)
+                # recode past Int64 values
+                intsent = uint64(INT_SENTINELS[intsentinel.x])
+                for i = 1:(row - 1)
+                    @inbounds z = tape[i]
+                    @inbounds tape[i] = ifelse(z === intsent, sentinelvalue(Float64), uint64(Float64(int64(z))))
```

Comment:
In the past, we didn't go back and recode already-parsed Int64 values to Float64, but stored a bit to say that it was originally a `Int64` or not. We now don't keep the extra `poslen` vectors around after parsing (since when we finish, we then know the perfect types for each column and can free that memory up).

Now, if we have been parsing an `Int64` column that needs to be promoted to `Float64`, we just go recode the past values, which makes less work in `CSV.Column` indexing and one less thing to keep track of.